### PR TITLE
feat(auth-connect): add the getting started guide

### PR DESCRIPTION
**Pull request by kensodemann**

### CODE REVIEW

**Review by eric-horodyski — COMMENTED** [review]
Now that we can display instructions and code side-by-side, does it make sense to trim down certain bits of instruction/explanation? For example:

> In the `capacitor.config.ts` file, change the `appId` to be something unique. The `appId` is used as the bundle ID. Therefore it should be a string that unique to your organization and application. We will use `io.ionic.gettingstartedangular` for this application. 

We could trim some of this paragraph since the reader would see the filename and changes reflected to their right:

> Change the `appId` to be something unique. The `appId` is used as the bundle ID. Therefore, it should be a string that is unique to your organization and application.

This is more food for thought than a request to make any changes.

**Review by eric-horodyski — APPROVED** [review]

File: docs/native-solutions/auth-connect/angular/introduction.md [hunk 1]
```diff
@@ -0,0 +1,1466 @@
...
+		<dict>
+			<key>CFBundleURLSchemes</key>
+			<array>
+				<string>msauth</string>
+			</array>
+		</dict>
+	</array>
+</dict>
+</plist>
+```
```
Comment (eric-horodyski) [inline docs/native-solutions/auth-connect/angular/introduction.md:778]: Wrap this in a `<CH.Code>` component and use the [rows](https://codehike.org/docs/ch-code#height) prop to adjust the height to just what the user needs highlighted. I tested with `<CH.Code rows={10}>` and it looks pretty good.

File: docs/native-solutions/auth-connect/angular/introduction.md [hunk 2]
```diff
@@ -0,0 +1,1466 @@
...
+        ]
+    }
+    buildTypes {
+        release {
+            minifyEnabled false
+            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
+        }
+    }
+}
+```
```
Comment (eric-horodyski) [inline docs/native-solutions/auth-connect/angular/introduction.md:753]: Wrap this in a `<CH.Code>` component and use the [rows](https://codehike.org/docs/ch-code#height) prop to adjust the height to just what the user needs highlighted. I tested with `<CH.Code rows={4}>` and it looks pretty good.

**Review by eric-horodyski — CHANGES_REQUESTED** [review]

File: docs/native-solutions/auth-connect/angular/introduction.md (hunk 1 above)
Comment (kensodemann) [reply docs/native-solutions/auth-connect/angular/introduction.md:778]: Nice! I was actually wondering if there was a way to do that then forgot to ask.

File: docs/native-solutions/auth-connect/angular/introduction.md (hunk 2 above)
Comment (kensodemann) [reply docs/native-solutions/auth-connect/angular/introduction.md]: Done

File: docs/native-solutions/auth-connect/angular/introduction.md (hunk 1 above)
Comment (kensodemann) [reply docs/native-solutions/auth-connect/angular/introduction.md]: done

**Comment by kensodemann** [comment]
> Now that we can display instructions and code side-by-side, does it make sense to trim down certain bits of instruction/explanation?

I trimmed that specific one. I was not sure if there were others you had in mind. In some cases I am doing this when we switch from one file to another in a multi-file set to make it more obvious that we have switched files.

Perhaps those should stay, not sure. Have a look.

If you have other specific cases you see, though, I am happy to address them.

File: docs/native-solutions/auth-connect/angular/introduction.md [hunk 3]
```diff
@@ -0,0 +1,1514 @@
...
+```typescript authentication.service.ts focus=9,20:21,24:25
+import { Injectable } from '@angular/core';
+import {
+  Auth0Provider,
+  AuthConnect,
+  AuthResult,
+  ProviderOptions,
+} from '@ionic-enterprise/auth';
+import { Platform } from '@ionic/angular';
+import { BehaviorSubject, Observable } from 'rxjs';
```
[the remark this answers is not in the document]
Comment (kensodemann) [reply docs/native-solutions/auth-connect/angular/introduction.md]: OK, I have fixed that... Switched to having an `isAuthenticated()`, which will work out better in the long run.

File: docs/native-solutions/auth-connect/angular/introduction.md [hunk 4]
```diff
@@ -0,0 +1,1511 @@
...
+```
+
+```bash Terminal focus=2
+ionic start getting-started-ac-angular tabs --type=angular-standalone
+cd getting-started-ac-angular
+```
+
+</CH.Code>
+
+Change the `appId` to be something unique. The `appId` is used as the [bundle ID](https://developer.apple.com/documentation/appstoreconnectapi/bundle_ids) / [application ID](https://developer.android.com/build/configure-app-module#set-application-id). Therefore it should be a string that unique to your organization and application. We will use `io.ionic.gettingstartedangular` for this application.
```
Comment (jameson-w-taylor) [inline docs/native-solutions/auth-connect/angular/introduction.md:62]: "...should be a string that unique...", should fix to say either "that's" or "that is"

File: docs/native-solutions/auth-connect/angular/introduction.md [hunk 5]
```diff
@@ -0,0 +1,1511 @@
...
+
+We need to perform a one-time setup with AuthConnect. Please refer to the [documentation](https://ionic.io/docs/auth-connect/interfaces/AuthConnectConfig) if you have any questions about the individual properties. We will start here with a simple set up that is good for development.
+
+The promise returned by `AuthConnect.setup()` is stored in our service so we can ensure the setup has completed before we execute code in methods we will add later.
+
+</CH.Scrollycoding>
+
+## Handling the Authentication Flow
+
+Auth Connect is now properly set up and initialized. We can move on to creating the basic log in and log out flow. Within this flow, and `AuthResult` is obtained during log in that represents our authentication session. So long as we have an `AuthResult` object, we have an authentication session. The `AuthResult` is no longer valid after the user logs out.
```
Comment (jameson-w-taylor) [inline docs/native-solutions/auth-connect/angular/introduction.md:320]: "...and `AuthResult` is..." should be "an"

File: docs/native-solutions/auth-connect/angular/introduction.md [hunk 6]
```diff
@@ -0,0 +1,1511 @@
...
+  async login(): Promise<void> {
+    await this.isReady;
+    this.authResult = await AuthConnect.login(this.provider, this.authOptions);
+  }
+}
+```
+
+</CH.Code>
+
+For the `login()`, we need to pass both the `provider` and the `options` we established earlier. Note that we for the `setup()` call to resolve and that we store the result in our session variable.
```
Comment (jameson-w-taylor) [inline docs/native-solutions/auth-connect/angular/introduction.md:395]: "...that we for the...", should be "we wait for"

File: docs/native-solutions/auth-connect/angular/introduction.md [hunk 7]
```diff
@@ -0,0 +1,1511 @@
...
+</CH.Scrollycoding>
+
+Test this in the web using the following credentials:
+
+- email: `test@ionic.io`
+- password: `Ion54321`
+
+At this point if we press the Login button, a tab should open where we can log in using Auth0. This tab will close after we log in. When we press the logout button a tab will briefly open to perform the logout and then automatically close.
+
+Note that if you press the Login button while already logged in the login tag is closed immediately. This is expected behavior.
```
Comment (jameson-w-taylor) [inline docs/native-solutions/auth-connect/angular/introduction.md:702]: "...the login tag is...", should be "tab"

File: docs/native-solutions/auth-connect/angular/introduction.md [hunk 8]
```diff
@@ -0,0 +1,1511 @@
...
+
+### Configure the Native Projects
+
+This login and logout are working on the web. Build your application for mobile and try to run them there. You can an emulator or an actual device for this test.
```
Comment (jameson-w-taylor) [inline docs/native-solutions/auth-connect/angular/introduction.md:706]: "...can an emulator or...", should be "can use an emulator"

File: docs/native-solutions/auth-connect/getting-started.md [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+
+For more information on OIDC, please see the Auth Connect documentation on [Understanding OpenID Connect](https://ionic.io/docs/auth-connect#addendum-understanding-openid-connect).
+
+## Goals
+
+In these tutorials, you will learn how to:
+
+- Configure Auth Connect based on the configuration of your Authentication vendor.
+- Set up and initialize Auth Connect within an application.
+- Create a basic authentication from from log in to log out.
```
Comment (jameson-w-taylor) [inline docs/native-solutions/auth-connect/getting-started.md:19]: "...authentication from from log...", remove extra "from"

File: docs/native-solutions/auth-connect/getting-started.md [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+- Manage the authentication session that is represented in Auth Connect via a `AuthResult` object.
```
Comment (jameson-w-taylor) [inline docs/native-solutions/auth-connect/getting-started.md:20]: "...via a `AuthResult`...", should be "an"

**Review by jameson-w-taylor — CHANGES_REQUESTED** [review]
Added some inline comments for a few grammatical fixes.

File: docs/native-solutions/auth-connect/angular/introduction.md (hunk 4 above)
Comment (kensodemann) [reply docs/native-solutions/auth-connect/angular/introduction.md:62]: Fixed

File: docs/native-solutions/auth-connect/angular/introduction.md (hunk 5 above)
Comment (kensodemann) [reply docs/native-solutions/auth-connect/angular/introduction.md:320]: fixed

File: docs/native-solutions/auth-connect/angular/introduction.md (hunk 6 above)
Comment (kensodemann) [reply docs/native-solutions/auth-connect/angular/introduction.md:395]: fixed

File: docs/native-solutions/auth-connect/angular/introduction.md (hunk 7 above)
Comment (kensodemann) [reply docs/native-solutions/auth-connect/angular/introduction.md:702]: Fixed

File: docs/native-solutions/auth-connect/getting-started.md (hunk 10 above)
Comment (kensodemann) [reply docs/native-solutions/auth-connect/getting-started.md:20]: Fixed

File: docs/native-solutions/auth-connect/getting-started.md [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+- `AzureProvider`
+- `CognitoProvider`
+- `OktaProvider`
+- `OneLoginProvider`
+
+You can also create your own custom provider, though doing so is beyond the scope of this tutorial.
+
+### The `ProviderOptions` Data
+
+When an authentication vendor, such as AWS Cognito, is configured, and application is created. This application includes several pieces of data that Auth Connect needs in order to communicate with it. These are specified via a `ProviderOptions` object. They include:
```
[the remark this answers is not in the document]
Comment (kensodemann) [reply docs/native-solutions/auth-connect/getting-started.md:44]: Fixed. Also, the use of commas seemed excessive in this context so I removed them, but I am also open to other opinions on that.

**Review by jameson-w-taylor — APPROVED** [review]

**Review by jameson-w-taylor — APPROVED** [review]

File: docs/native-solutions/auth-connect/angular/introduction.md [hunk 12]
```diff
@@ -0,0 +1,1511 @@
...
+```
+
+```bash Terminal focus=2
+ionic start getting-started-ac-angular tabs --type=angular-standalone
+cd getting-started-ac-angular
+```
+
+</CH.Code>
+
+Change the `appId` to be something unique. The `appId` is used as the [bundle ID](https://developer.apple.com/documentation/appstoreconnectapi/bundle_ids) / [application ID](https://developer.android.com/build/configure-app-module#set-application-id). Therefore it should be a string that is unique to your organization and application. We will use `io.ionic.gettingstartedangular` for this application.
```
Comment (eric-horodyski) [inline docs/native-solutions/auth-connect/angular/introduction.md:62]: This text uses `io.ionic.gettingstartedangular` as an App ID, but the code example uses `io.ionic.gettingstartedac`.

File: docs/native-solutions/auth-connect/angular/introduction.md [hunk 13]
```diff
@@ -0,0 +1,1511 @@
...
+      redirectUri: isNative ? 'msauth://login' : 'http://localhost:8100/login',
+      scope: 'openid offline_access email picture profile',
+    };
+  }
+}
+```
+
+</CH.Code>
+
+Auth Connect needs to know how to communicate with our Authentication vendor. You will likely need to get this information from the team that manages your cloud infrastructure.
```
Comment (eric-horodyski) [inline docs/native-solutions/auth-connect/angular/introduction.md:259]: Remove capitalization from the word "Authentication":

> Auth Connect needs to know how to communicate with our authentication vendor.

File: docs/native-solutions/auth-connect/angular/introduction.md [hunk 14]
```diff
@@ -0,0 +1,1511 @@
...
+        authFlow: 'implicit',
+      },
+    });
+  }
+}
+```
+
+</CH.Code>
+
+We need to perform a one-time setup with AuthConnect. Please refer to the [documentation](https://ionic.io/docs/auth-connect/interfaces/AuthConnectConfig) if you have any questions about the individual properties. We will start here with a simple set up that is good for development.
```
Comment (eric-horodyski) [inline docs/native-solutions/auth-connect/angular/introduction.md:312]: Make "AuthConnect" two words:

> We need to perform a one-time setup with Auth Connect.

File: docs/native-solutions/auth-connect/angular/introduction.md [hunk 15]
```diff
@@ -0,0 +1,1511 @@
...
+
+</CH.Code>
+
+Replace it with a couple of buttons.
+
+You can also remove any references to `ExploreContainerComponent` in `tabs1.page.ts`.
+
+</CH.Scrollycoding>
+
+Test this in the web using the following credentials:
```
Comment (eric-horodyski) [inline docs/native-solutions/auth-connect/angular/introduction.md:695]: > Test this in the web

Change to "Test this _on_ the web" or "Test this in _your_ web _browser_"

File: docs/native-solutions/auth-connect/angular/introduction.md [hunk 16]
```diff
@@ -0,0 +1,1511 @@
...
+- email: `test@ionic.io`
+- password: `Ion54321`
+
+At this point if we press the Login button, a tab should open where we can log in using Auth0. This tab will close after we log in. When we press the logout button a tab will briefly open to perform the logout and then automatically close.
+
+Note that if you press the Login button while already logged in the login tab is closed immediately. This is expected behavior.
+
+### Configure the Native Projects
+
+This login and logout are working on the web. Build your application for mobile and try to run them there. You can use an emulator or an actual device for this test.
```
Comment (eric-horodyski) [inline docs/native-solutions/auth-connect/angular/introduction.md:706]: Remove the word "This":

> "Login and logout are working on the web."

File: docs/native-solutions/auth-connect/angular/introduction.md [hunk 17]
```diff
@@ -0,0 +1,1511 @@
...
+
+</CH.Code>
+
+If we have an `AuthResult` we will assume that we are authenticated. The authentication session _could_ be expired or otherwise invalid, but we will work on handling that in other tutorials.
+
+---
+
+<CH.Code>
+
+```typescript tab1.page.ts focus=19,24,27:29
```
Comment (eric-horodyski) [inline docs/native-solutions/auth-connect/angular/introduction.md:956]: Add highlighting to LN13.

**Review by eric-horodyski — CHANGES_REQUESTED** [review]
Looking really good, just a few grammatical changes and one highlighting change.

File: docs/native-solutions/auth-connect/angular/introduction.md (hunk 12 above)
Comment (kensodemann) [reply docs/native-solutions/auth-connect/angular/introduction.md:62]: fixed

File: docs/native-solutions/auth-connect/angular/introduction.md (hunk 13 above)
Comment (kensodemann) [reply docs/native-solutions/auth-connect/angular/introduction.md:259]: done

File: docs/native-solutions/auth-connect/angular/introduction.md (hunk 14 above)
Comment (kensodemann) [reply docs/native-solutions/auth-connect/angular/introduction.md:312]: Done

File: docs/native-solutions/auth-connect/angular/introduction.md (hunk 15 above)
Comment (kensodemann) [reply docs/native-solutions/auth-connect/angular/introduction.md:695]: done

File: docs/native-solutions/auth-connect/angular/introduction.md (hunk 16 above)
Comment (kensodemann) [reply docs/native-solutions/auth-connect/angular/introduction.md:706]: done

File: docs/native-solutions/auth-connect/angular/introduction.md (hunk 17 above)
Comment (kensodemann) [reply docs/native-solutions/auth-connect/angular/introduction.md:956]: done

**Review by eric-horodyski — APPROVED** [review]

**Pull request merged**